moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1713,6 +1713,14 @@ bool PlanningScene::processCollisionObjectMsg(const moveit_msgs::CollisionObject
   return false;
 }
 
+void PlanningScene::poseMsgToEigen(const geometry_msgs::Pose& msg, Eigen::Isometry3d& out)
+{
+  Eigen::Translation3d translation(msg.position.x, msg.position.y, msg.position.z);
+  Eigen::Quaterniond quaternion(msg.orientation.w, msg.orientation.x, msg.orientation.y, msg.orientation.z);
+  quaternion.normalize();
```

Comment:
Shouldn't this be added to `tf2_eigen.h` or even all conversion functions with quaternions? `fromMsg()` could have a default parameter `normalize_quaternion=false`.

Reply:
This would be indeed a nice to have. However, OSRF wasn't very responsive for ROS1 changes in the recent past.

Reply:
I can change the calls to tf2::fromMsg() in other places, however I am not so sure I this is really desired as then quaternion normlization is done more often. I read up on the discussion [here](https://github.com/ros-visualization/rviz/pull/1179), and for me it sounded like we don't want to normalize the quaternions everywhere. But maybe @rhaschke can clarify that?

Reply:
If that's the case it's fine with me. It just seems odd that collision objects are affected while attached collision objects are not. I approve this patch as is. @rhaschke do you have something to add here? Otherwise lgtm.

Reply:
I looked into this briefly a week ago and I agree that we should consider normalization in more places. Investigating the extent of required changes, I found some nasty code duplication between `PlanningScene` code and `robot_state/.../conversions.cpp`. I wanted to clean this up but didn't find time yet to do so.

Reply:
@henningkayser, let's merge for now. Remember to cherry-pick to master. This PR is targeted at melodic-devel currently.